src/overlays/actors/ovl_En_Bom_Chu/z_en_bom_chu.c

```diff
@@ -129,61 +131,72 @@ void EnBomChu_CrossProduct(Vec3f* a, Vec3f* b, Vec3f* dest) {
     dest->z = (a->x * b->y) - (a->y * b->x);
 }
 
-void func_809C5BA8(EnBomChu* this, CollisionPoly* floorPoly, GlobalContext* globalCtx) {
-    Vec3f sp84;
-    Vec3f sp78;
-    f32 sp74;
+void EnBomChu_UpdateFloorPoly(EnBomChu* this, CollisionPoly* floorPoly, GlobalContext* globalCtx) {
+    Vec3f normal;
+    Vec3f vec;
+    f32 angle;
     f32 magnitude;
-    f32 temp_f12;
+    f32 normDotUp;
     MtxF mf;
 
     this->actor.floorPoly = floorPoly;
 
-    sp84.x = COLPOLY_GET_NORMAL(floorPoly->normal.x);
-    sp84.y = COLPOLY_GET_NORMAL(floorPoly->normal.y);
-    sp84.z = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+    normal.x = COLPOLY_GET_NORMAL(floorPoly->normal.x);
+    normal.y = COLPOLY_GET_NORMAL(floorPoly->normal.y);
+    normal.z = COLPOLY_GET_NORMAL(floorPoly->normal.z);
+
+    normDotUp = (normal.x * this->axisUp.x) + (normal.y * this->axisUp.y) + (normal.z * this->axisUp.z);
```

Comment:
```suggestion
    normDotUp = DOTXYZ(normal, this->axisUp);
```
Just something I noticed